### PR TITLE
MOSIP-43501: Add expiryPeriod optional query param in GET /misp-license API

### PR DESCRIPTION
**Pull request by SradhaMohanty5899**
Automated test cases for validating the optional expiryPeriod query parameter in GET /misp-license API, covering scenarios with valid and invalid values, including status code verification, and error handling.

## Summary by CodeRabbit

* **New Features**
  * Deactivate MISP licenses and partners; update partner API key expiry date flows.
  * Enhanced license listing: input-driven pagination, sorting, status and expiry filters.

* **Tests**
  * Many new positive and negative test scenarios for deactivation, expiry updates, license queries and notifications.
  * Expanded edge-case coverage, some test identifiers renamed, expiry dates parameterized.

* **Other**
  * Test/template data tweaks, broader delete patterns, simplified responses, and date-input normalization.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: api-test/src/main/resources/pms/DeactivateMISPLicense/DeactivateMISPLicenseResult.hbs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "response": {
+    "partnerId": "$IGNORE$",
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/DeactivateMISPLicense/DeactivateMISPLicenseResult.hbs]: Why ignore parameters are added newly?

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml [hunk 2]
```diff
@@ -39,9 +39,9 @@ GetMISPPartnerLicense:
         output: '{
 }'
     Pms_GetMISPPartnerLicense_with_Status_Activated:
-        endPoint: /v1/partnermanager/misp-licenses?status=ACTIVE
+        endPoint: /v1/partnermanager/misp-licenses?status=activated
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml]: why the query param is modified?

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml:1]: Why all the existing params are modified?

Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml:1]: In the newly added testcases how you are making sure, endpoints are working as there is not output validation done?

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml [hunk 3]
```diff
@@ -168,3 +168,211 @@ GetMISPPartnerLicense:
...
+        restMethod: get
+        checkErrorsOnlyInResponse: true
+        inputTemplate: pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense
+        outputTemplate: pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseResult
+        input: '{
+}'
+        output: '{
+}'
+    Pms_GetMISPPartnerLicense_With_PartnerId_And_PolicyName:
+        endPoint: /v1/partnermanager/misp-licenses?partnerId=pms-111555&policyName=mosip policy MISP API Policy 12345
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml]: Policy name and partner id should not be hardcoded

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml [hunk 4]
```diff
@@ -168,3 +168,211 @@ GetMISPPartnerLicense:
...
+        restMethod: get
+        checkErrorsOnlyInResponse: true
+        inputTemplate: pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense
+        outputTemplate: pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseResult
+        input: '{
+}'
+        output: '{
+}'
+    Pms_GetMISPPartnerLicense_With_Numeric_LicenseKeyName:
+        endPoint: /v1/partnermanager/misp-licenses?mispLicenseKeyName=1234
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml]: license key should not be hardcoded

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml [hunk 5]
```diff
@@ -352,4 +352,261 @@ GetMISPPartnerLicenseNegTC:
...
+    uniqueIdentifier: TC_PMS_GetMISPPartnerLicenseNegTC_25
+    description: Fetching MISP Partner License By providing Negative ExpiryPeriod
+    role: partneradmin
+    restMethod: get
+    checkOnlyStatusCodeInResponse: true
+    inputTemplate: pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense
+    outputTemplate: pms/responseCode
+    input: '{}'
+    output: '{
+    "responseCode": "400"
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml:661]: Why error handling not done for these from application side?

**Review by mohanachandran-s — CHANGES_REQUESTED** [review]
1. Remove the hardcoded values in the code..
2. added the possible output validations for the missing

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml:1]: Previously we are passing some UI data so It can be failed in other new env so this time passed everything from automation

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml (hunk 2 above)
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml]: The right word is activated, with ACTIVE showing no records

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml (hunk 5 above)
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml:661]: They have done control level changes means data type validation, if it will be not same datatype then 400 bad request will some

File: api-test/src/main/resources/pms/DeactivateMISPLicense/DeactivateMISPLicenseResult.hbs (hunk 1 above)
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/DeactivateMISPLicense/DeactivateMISPLicenseResult.hbs]: Removed

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense.yml:1]: We can only validate using pageSize, pageNo and totalResults but it can be varied in large dataset so validation should not be there.

File: api-test/src/main/resources/pms/UpdateApiKeyExpiryDate/UpdateApiKeyExpiryDateNegTC.yml [hunk 6]
```diff
@@ -0,0 +1,515 @@
...
+      "id": "mosip.pms.update.api.key.expiry.patch",
+      "version": "1.0",
+      "requestTime": "$TIMESTAMP$",
+      "partnerId": "pms-111999",
+      "policyId": "3456%",
+      "apiKeyName": "$ID:GenerateApiKey_AllValid_smoke_sid_label$",
+      "apiKeyExpiryDateTime": "$EXPIRYDATE:365$"
+}'
+    output: '{
+      "responseCode": "400"
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/UpdateApiKeyExpiryDate/UpdateApiKeyExpiryDateNegTC.yml]: Checked

File: api-test/src/main/resources/pms/UpdateApiKeyExpiryDate/UpdateApiKeyExpiryDateNegTC.yml [hunk 7]
```diff
@@ -0,0 +1,515 @@
+UpdateApiKeyExpiryDateNegTC:
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/UpdateApiKeyExpiryDate/UpdateApiKeyExpiryDateNegTC.yml:1]: Checked

File: api-test/src/main/resources/pms/UpdateApiKeyExpiryDate/UpdateApiKeyExpiryDate.yml [hunk 8]
```diff
@@ -0,0 +1,61 @@
+UpdateApiKeyExpiryDate:
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/UpdateApiKeyExpiryDate/UpdateApiKeyExpiryDate.yml:1]: Checked

File: api-test/src/main/java/io/mosip/testrig/apirig/partner/testscripts/SimplePostForAutoGenId.java [hunk 9]
```diff
@@ -86,6 +86,9 @@ public void test(TestCaseDTO testCaseDTO)
 			throws AuthenticationTestException, AdminTestException, NoSuchAlgorithmException, SecurityXSSException {
 		testCaseName = testCaseDTO.getTestCaseName();
 		testCaseName = PMSUtil.isTestCaseValidForExecution(testCaseDTO);
+		if(testCaseDTO.getTestCaseName().contains("UploadPartnerCert_All_Valid_Smoke")) {
+			System.out.println("Debug Point");
+		}
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/java/io/mosip/testrig/apirig/partner/testscripts/SimplePostForAutoGenId.java]: Reverted

File: api-test/src/main/java/io/mosip/testrig/apirig/partner/testscripts/PatchWithPathParamsAndBody.java [hunk 10]
```diff
@@ -81,14 +81,17 @@ public Object[] getTestCaseList(ITestContext context) {
 	public void test(TestCaseDTO testCaseDTO) throws AdminTestException, SecurityXSSException {
 		testCaseName = testCaseDTO.getTestCaseName();
 		testCaseName = PMSUtil.isTestCaseValidForExecution(testCaseDTO);
+		if(testCaseDTO.getUniqueIdentifier().contains("TC_PMS_UpdateApiKeyExpiryDate_01")) {
+			System.out.println("Debug point");
+		}
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/java/io/mosip/testrig/apirig/partner/testscripts/PatchWithPathParamsAndBody.java]: Reverted

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml [hunk 11]
```diff
@@ -352,4 +352,261 @@ GetMISPPartnerLicenseNegTC:
...
+        inputTemplate: pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicense
+        outputTemplate: pms/error
+        input: '{}'
+        output: '{
+        "errors": [
+           {
+              "errorCode": "PMS_PS_ERROR_007"
+           }
+        ]
+}'
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml:637]: Resolved

File: api-test/src/main/resources/pms/UpdateApiKeyExpiryDate/UpdateApiKeyExpiryDate.yml [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+      "id": "mosip.pms.update.api.key.expiry.patch",
+      "version": "1.0",
+      "requestTime": "$TIMESTAMP$",
+      "partnerId": "pms-111999",
+      "policyId": "$ID:GenerateApiKeysforApprovedPolicies_AllValid_smoke_sid_policyId$",
+      "apiKeyName": "$ID:GenerateApiKeysforApprovedPolicies_AllValid_smoke_sid_label$",
+      "apiKeyExpiryDateTime": "$EXPIRYDATE_10YEARS$"
+}'
+    output: '{
+}'
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/UpdateApiKeyExpiryDate/UpdateApiKeyExpiryDate.yml:61]: These dependent testcases are available in the module.

File: api-test/src/main/resources/config/partnerRevampDataDeleteQueries.txt [hunk 13]
```diff
@@ -10,6 +10,7 @@ delete from pms.partner_contact where cr_by='pms-111997';
 delete from partner_policy_bioextract where cr_by='pms-111997';
 delete from pms.misp_license where cr_by='pms-111997';
 delete from pms.misp_license where license_key_name like 'MISP_API_Automation%';
+delete from pms.misp_license where license_key_name='1234';
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/config/partnerRevampDataDeleteQueries.txt]: Modify the delete query to delete using cr_by

File: api-test/src/main/java/io/mosip/testrig/apirig/partner/testscripts/GetWithQueryParam.java [hunk 14]
```diff
@@ -89,7 +89,10 @@ public void test(TestCaseDTO testCaseDTO) throws AuthenticationTestException, Ad
 		}
 
 		String[] templateFields = testCaseDTO.getTemplateFields();
-
+		String inputJson = getJsonFromTemplate(testCaseDTO.getInput(), testCaseDTO.getInputTemplate());
+		inputJson = inputJsonKeyWordHandeler(inputJson, testCaseName);
+		inputJson = normalizeDateFields(inputJson);
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/java/io/mosip/testrig/apirig/partner/testscripts/GetWithQueryParam.java:94]: Removed

File: api-test/src/main/resources/pms/GetAllNotifications/GetAllNotifications.yml [hunk 15]
```diff
@@ -636,4 +636,359 @@ GetAllNotifications:
...
+      outputTemplate: pms/error
+      input: '{
+}'
+      output: '{
+      "errors": [
+      {
+       "errorCode": "KER-ATH-401"
+       }
+       ]
+}'
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/GetAllNotifications/GetAllNotifications.yml:656]: Removed

File: api-test/src/main/resources/pms/MISPPartnerV3/UploadMISPPartnerCertificate/UploadMISPPartnerCertificate.yml [hunk 16]
```diff
@@ -103,4 +103,18 @@ UploadMISPPartnerCertificate:
...
+      role: partneradmin
+      checkErrorsOnlyInResponse: true
+      restMethod: post
+      inputTemplate: pms/MISPPartnerV3/UploadMISPPartnerCertificate/UploadMISPPartnerCertificate112211
+      outputTemplate: pms/MISPPartnerV3/UploadMISPPartnerCertificate/UploadMISPPartnerCertResult
+      input: '{
+  "requesttime": "$TIMESTAMP$"
+}'
+      output: ' {
 }'
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/MISPPartnerV3/UploadMISPPartnerCertificate/UploadMISPPartnerCertificate.yml:121]: Added

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml [hunk 17]
```diff
@@ -346,10 +610,495 @@ GetMISPPartnerLicenseNegTC:
...
+    "policyGroupName": "$REMOVE$",
+    "policyName": "$REMOVE$",
+    "mispLicenseKeyName": "$REMOVE$",
+    "sortFieldName": "$REMOVE$",
+    "sortType": "$REMOVE$",
+    "status": "$REMOVE$",
+    "expiryPeriod": "$REMOVE$"
+}'
+    output: '{
 }'
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml:1082]: Resolved

File: api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml [hunk 18]
```diff
@@ -16,29 +28,53 @@ GetMISPPartnerLicenseNegTC:
...
+		"partnerId": "$REMOVE$",
+		"orgName": "Par@@!!$$.",
+		"policyGroupName": "$REMOVE$",
+		"policyName": "$REMOVE$",
+		"mispLicenseKeyName": "$REMOVE$",
+		"sortFieldName": "$REMOVE$",
+		"sortType": "$REMOVE$",
+		"status": "$REMOVE$",
+		"expiryPeriod": "$REMOVE$"
+	}'
```
[the remark this answers is not in the document]
Comment (SradhaMohanty5899) [reply api-test/src/main/resources/pms/MispLicenseV3/GetMISPPartnerLicense/GetMISPPartnerLicenseNegTC.yml:77]: Resolved

File: api-test/src/main/resources/pms/GetAllNotifications/GetAllNotifications.yml [hunk 19]
```diff
@@ -636,4 +636,359 @@ GetAllNotifications:
...
+   Pms_GetAllNotifications_with_Apikey_NotificationType_And_CertificateId_Neg:
+      endPoint: /v1/partnermanager/notifications
+      role: partneradmin
+      description: Get All Notifications with API Key Notification Type and Certificate ID
+      uniqueIdentifier: TC_PMS_GetAllNotifications_50
+      restMethod: get
+      checkErrorsOnlyInResponse: true
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/GetAllNotifications/GetAllNotifications.yml]: checkErrorsOnlyInResponse: true should not be used for negative testcases

File: api-test/src/main/resources/pms/MispLicenseV3/GenerateMISPPartnerLicense/GenerateMISPPartnerLicense.yml [hunk 20]
```diff
@@ -35,9 +35,93 @@ GenerateMISPPartnerLicense:
...
+      description: Generate MISP Partner License without PolicyId
+      role: partneradmin
+      restMethod: post
+      inputTemplate: pms/MispLicenseV3/GenerateMISPPartnerLicense/GenerateMISPPartnerLicense
+      outputTemplate: pms/MispLicenseV3/GenerateMISPPartnerLicense/GenerateMISPPartnerLicenseResult
+      input: '{
+        "id": "mosip.pms.misp.generate.license.post",
+        "version": "1.0",
+        "requestTime": "$TIMESTAMP$",
+        "partnerId": "pms-112211",
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/MispLicenseV3/GenerateMISPPartnerLicense/GenerateMISPPartnerLicense.yml]: Instead of hardcoding, manage it with auto generated ID

**Review by mohanachandran-s — CHANGES_REQUESTED** [review]

**Review by mohanachandran-s — APPROVED** [review]

**Review by mayuradesh — APPROVED** [review]
